test/local_integration_test.py

```diff
@@ -264,7 +298,32 @@ def check_bundles_are_indexed(self, test_name: str, entity_type: str, indexed_bu
                             f' {hit["entryId"]}. Bundle(s) ({",".join(bundle_fqids)})'
                             f' have been indexed without the debug project name. Contains {project}')
 
-    @classmethod
-    def _get_file_uuid(cls, response: bytes) -> str:
-        """Returns file UUID of first FASTQ file."""
-
+    def project_removed_from_azul(self):
+        results_empty = [len(self._get_entities_by_project(entity, self.test_name)) == 0
+                         for entity in ['files', 'projects', 'samples']]
+        logger.info("Project removed from index files: {}, projects: {}, specimens: {}".format(*results_empty))
+        return all(results_empty)
+
+    def _get_entities_by_project(self, entity_type, project_shortname):
+        """
+        Returns all entities in a given project.
+        """
+        filters = {'file': {'project': {'is': [project_shortname]}}}
+        files = []
+        size = 100
+        params = dict(filters=str(filters), size=str(size))
+        while True:
```

Comment:
There is similar code in `check_bundles_are_indexed` and I think we can't afford to maintain them alongside each other. Could you please consolidate these two? `check_bundles_are_indexed` also uses a weird timeout which I think is unnecessary.

On a general note: As long as I am project lead, I will pursue keeping entropy and duplication low in the code base unless there is a good reason not do do so, like time pressure. While I understand the desire to churn thru tickets and check boxes, our main concern should be long-term maintainability and minimization of technical debt.

Reply:
I appreciate the desire to refactor. Will do.